_studio/shared/include/mfx_utils.h

```diff
@@ -37,33 +37,13 @@
 #include <va/va.h>
 #endif
 
-#ifndef MFX_DEBUG_TRACE
-#define MFX_STS_TRACE(sts) sts
-#else
-template <typename T>
-static inline T mfx_print_err(T sts, const char *file, int line, const char *func)
-{
-    if (sts)
-    {
-        printf("%s: %d: %s: Error = %d\n", file, line, func, sts);
-    }
-    return sts;
-}
-#define MFX_STS_TRACE(sts) mfx_print_err(sts, __FILE__, __LINE__, __FUNCTION__)
-#endif
+#include "mfx_utils_defs.h"
 
-#define MFX_SUCCEEDED(sts)      (MFX_STS_TRACE(sts) == MFX_ERR_NONE)
-#define MFX_FAILED(sts)         (MFX_STS_TRACE(sts) != MFX_ERR_NONE)
-#define MFX_RETURN(sts)         { return MFX_STS_TRACE(sts); }
-#define MFX_CHECK(EXPR, ERR)    { if (!(EXPR)) MFX_RETURN(ERR); }
 
 #define MFX_CHECK_NO_RET(EXPR, STS, ERR){ if (!(EXPR)) { std::ignore = MFX_STS_TRACE(ERR); STS = ERR; } }
 
-#define MFX_CHECK_STS(sts)              MFX_CHECK(MFX_SUCCEEDED(sts), sts)
 #define MFX_SAFE_CALL(FUNC)             { mfxStatus _sts = FUNC; MFX_CHECK_STS(_sts); }
-#define MFX_CHECK_NULL_PTR1(pointer)    MFX_CHECK(pointer, MFX_ERR_NULL_PTR)
-#define MFX_CHECK_NULL_PTR2(p1, p2)     { MFX_CHECK(p1, MFX_ERR_NULL_PTR); MFX_CHECK(p2, MFX_ERR_NULL_PTR); }
-#define MFX_CHECK_NULL_PTR3(p1, p2, p3) { MFX_CHECK(p1, MFX_ERR_NULL_PTR); MFX_CHECK(p2, MFX_ERR_NULL_PTR); MFX_CHECK(p3, MFX_ERR_NULL_PTR); }
+
```

Comment:
I still don't see the point to move only part of these definitions to mfx_utils_defs.h. I believe you just need to move all of these defs to this new file regardless whether you use them in a new code or not.

Reply:
done